src/main/java/org/carlmontrobotics/robotcode2023/subsystems/Drivetrain.java

```diff
@@ -0,0 +1,354 @@
+            Supplier<Float> rollSupplier = () -> initRoll;
+            // initPitch = gyro.getPitch();
+            // initRoll = gyro.getRoll();
+            // Supplier<Float> pitchSupplier = () -> gyro.getPitch();
+            // Supplier<Float> rollSupplier = () -> gyro.getRoll();
+
+            SwerveModule moduleFL = new SwerveModule(swerveConfig, SwerveModule.ModuleType.FL,
+                    MotorControllerFactory.createSparkMax(driveFrontLeftPort, TemperatureLimit.NEO),
+                    MotorControllerFactory.createSparkMax(turnFrontLeftPort, TemperatureLimit.NEO),
+                    MotorControllerFactory.createCANCoder(canCoderPortFL), 0,
+                    pitchSupplier, rollSupplier);
+            // Forward-Right
+            SwerveModule moduleFR = new SwerveModule(swerveConfig, SwerveModule.ModuleType.FR,
+                    MotorControllerFactory.createSparkMax(driveFrontRightPort, TemperatureLimit.NEO),
+                    MotorControllerFactory.createSparkMax(turnFrontRightPort, TemperatureLimit.NEO),
+                    MotorControllerFactory.createCANCoder(canCoderPortFR), 1,
+                    pitchSupplier, rollSupplier);
+            // Backward-Left
+            SwerveModule moduleBL = new SwerveModule(swerveConfig, SwerveModule.ModuleType.BL,
+                    MotorControllerFactory.createSparkMax(driveBackLeftPort, TemperatureLimit.NEO),
+                    MotorControllerFactory.createSparkMax(turnBackLeftPort, TemperatureLimit.NEO),
+                    MotorControllerFactory.createCANCoder(canCoderPortBL), 2,
+                    pitchSupplier, rollSupplier);
+            // Backward-Right
+            SwerveModule moduleBR = new SwerveModule(swerveConfig, SwerveModule.ModuleType.BR,
+                    MotorControllerFactory.createSparkMax(driveBackRightPort, TemperatureLimit.NEO),
+                    MotorControllerFactory.createSparkMax(turnBackRightPort, TemperatureLimit.NEO),
+                    MotorControllerFactory.createCANCoder(canCoderPortBR), 3,
+                    pitchSupplier, rollSupplier);
+            modules = new SwerveModule[] { moduleFL, moduleFR, moduleBL, moduleBR };
+        }
+        SmartDashboard.putNumber("kpTheta", thetaPIDController[0]);
+        odometry = new SwerveDrivePoseEstimator(kinematics, Rotation2d.fromDegrees(getHeading()), getModulePositions(), new Pose2d());
+        prevTime = Timer.getFPGATimestamp();
+    }
+
+    @Override
+    public void periodic() {
+        for (SwerveModule module : modules) module.periodic();
+
+        // Update the odometry with current heading and encoder position
+        odometry.update(Rotation2d.fromDegrees(getHeading()), getModulePositions());
+
+        // Update the odometry with limelight
+        if(lime != null && lime.hasTarget()) {
+            odometry.addVisionMeasurement(lime.getTransform(Limelight.Transform.BOTPOSE).toPose2d(), WPIUtilJNI.now() * 1.0e-6 /* From the odometry.update implementation */);
```

Comment:
The last parameter passed to `addVisionMeasurement()` here is incorrect. See here: https://docs.limelightvision.io/en/latest/apriltags_in_3d.html#using-wpilib-s-pose-estimator

Reply:
My bad; I missed that! Should be fixed :D